resources/views/fields/upload.blade.php

```diff
@@ -85,7 +85,7 @@
                                 </div>
 
 
-                                @if($storage === 'public')
+                                @if(config('filesystems')['disks'][$storage ?? 'public']['visibility'] ?? 'private' === 'public')
```

Comment:
I think we should move this to the field class. And then only use the result.

```php
    /**
     * All attributes that are available to the field.
     *
     * @var array
     */
    protected $attributes = [
        // ...
        'visibility'  => 'public',
    ];
```

Then we would only check small nesting in view files:

```blade
@if($visibility === 'public')
   {{--  code --}}
@endif
```

And they would check already in the method:

```php
/**
 * @param string $storage
 *
 * @return $this
 * @throws \Throwable
 */
public function storage(string $storage) : self
{
    $disks = config('filesystems.disks');

    throw_unless(
        array_key_exists($storage, $disks),
        \Exception::class,
        'The selected storage was not found'
    );

    $this->set('storage', $storage);
    $this->set('visibility', Arr::get($disks, "$storage.visibility"));
}
```

What do you think about it?

Reply:
Or you can write it even shorter.

```php
public function storage(string $storage) : self
{
    $disk = config("filesystems.disks.$storage");

    throw_if(
        $storage === null,
        \Exception::class,
        'The selected storage was not found'
    );

    return $this
        ->set('storage', $storage)
        ->set('visibility', $disk['visibility'] ?? null);
}
```

Reply:
That's a much better solution, love it :+1: 

Would you like me to implement or are you happy to change the pull request?

Reply:
Please prepare a PR, with such a solution, as well as tests. If you have any difficulties, I will help you.

Reply:
@tabuna Will put on tonight todo list :+1: 

Thanks for the pointers, very helpful and good to start understanding how the platform works.